generator/internal/discovery_to_proto_test.cc

```diff
@@ -1651,16 +1651,22 @@ TEST_F(AssignResourcesAndTypesToFilesTest,
   auto result =
       AssignResourcesAndTypesToFiles(resources, types, props, "output_path");
   ASSERT_STATUS_OK(result);
-  ASSERT_THAT(result->size(), Eq(2));
+  ASSERT_THAT(result->first.size(), Eq(2));
   EXPECT_THAT(
-      (*result)[0].file_path(),
+      result->first[0].file_path(),
       Eq("output_path/google/cloud/product_name/foos/version/foos.proto"));
-  EXPECT_THAT((*result)[0].types(), IsEmpty());
-  EXPECT_THAT((*result)[1].file_path(),
+  EXPECT_THAT(result->first[0].types(), IsEmpty());
+  EXPECT_THAT(result->first[1].file_path(),
               Eq("output_path/google/cloud/product_name/"
                  "version/internal/common_000.proto"));
-  ASSERT_THAT((*result)[1].types().size(), Eq(1));
-  EXPECT_THAT((*result)[1].types().front()->name(), Eq("Operation"));
+  ASSERT_THAT(result->first[1].types().size(), Eq(1));
+  EXPECT_THAT(result->first[1].types().front()->name(), Eq("Operation"));
+
+  ASSERT_THAT(result->second, SizeIs(1));
```

Comment:
Use `ElementsAre()` to avoid the need for size assertions before testing values?